From 4cacfd1d519092a74aa9a88b2944aebe9cd7e7af Mon Sep 17 00:00:00 2001
From: Nikhil Agarwal <Nikhil.Agarwal@freescale.com>
Date: Tue, 12 Aug 2014 15:00:36 +0530
Subject: [PATCH 161/399] ASFIPSEC: Porting ASF to LS1.

Adding PDB endianness independent.

Signed-off-by: Nikhil Agarwal <Nikhil.Agarwal@freescale.com>
Change-Id: I44bfc921ba1460dd2785c21252898b55953d8385
Reviewed-on: http://git.am.freescale.net:8181/17497
Tested-by: Review Code-CDREVIEW <CDREVIEW@freescale.com>
Reviewed-by: Sandeep Malik <Sandeep.Malik@freescale.com>
Reviewed-by: Zhengxiong Jin <Jason.Jin@freescale.com>
Tested-by: Zhengxiong Jin <Jason.Jin@freescale.com>
(cherry picked from commit 16e2ab4fab053e020d4857f14fe8bd35616ad304)
Reviewed-on: http://git.am.freescale.net:8181/21354
[Lu:Original patch taken from
QorIQ-SDK-V1.7-SOURCE-20141218-yocto.iso]
Signed-off-by: Jiang Lu <lu.jiang@windriver.com>
---
 drivers/crypto/caam/pdb.h |   30 ++++++++++++++++++++++++++++++
 1 files changed, 30 insertions(+), 0 deletions(-)

diff --git a/drivers/crypto/caam/pdb.h b/drivers/crypto/caam/pdb.h
index aca22f9..7087be0 100644
--- a/drivers/crypto/caam/pdb.h
+++ b/drivers/crypto/caam/pdb.h
@@ -66,9 +66,15 @@ struct ipsec_encap_ctr {
 
 struct ipsec_encap_ccm {
 	u32 salt; /* lower 24 bits */
+#ifdef CONFIG_CPU_BIG_ENDIAN
 	u8 b0_flags;
 	u8 ctr_flags;
 	u16 ctr_initial;
+#else
+	u16 ctr_initial;
+	u8 ctr_flags;
+	u8 b0_flags;
+#endif
 	u32 iv[2];
 };
 
@@ -79,10 +85,17 @@ struct ipsec_encap_gcm {
 };
 
 struct ipsec_encap_pdb {
+#ifdef CONFIG_CPU_BIG_ENDIAN
 	u8 hmo_rsvd;
 	u8 ip_nh;
 	u8 ip_nh_offset;
 	u8 options;
+#else
+	u8 options;
+	u8 ip_nh_offset;
+	u8 ip_nh;
+	u8 hmo_rsvd;
+#endif
 	u32 seq_num_ext_hi;
 	u32 seq_num;
 	union {
@@ -92,8 +105,13 @@ struct ipsec_encap_pdb {
 		struct ipsec_encap_gcm gcm;
 	};
 	u32 spi;
+#ifdef CONFIG_CPU_BIG_ENDIAN
 	u16 rsvd1;
 	u16 ip_hdr_len;
+#else
+	u16 ip_hdr_len;
+	u16 rsvd1;
+#endif
 	u32 ip_hdr[0]; /* optional IP Header content */
 };
 
@@ -108,9 +126,15 @@ struct ipsec_decap_ctr {
 
 struct ipsec_decap_ccm {
 	u32 salt;
+#ifdef CONFIG_CPU_BIG_ENDIAN
 	u8 iv_flags;
 	u8 ctr_flags;
 	u16 ctr_initial;
+#else
+	u16 ctr_initial;
+	u8 ctr_flags;
+	u8 iv_flags;
+#endif
 };
 
 struct ipsec_decap_gcm {
@@ -119,9 +143,15 @@ struct ipsec_decap_gcm {
 };
 
 struct ipsec_decap_pdb {
+#ifdef CONFIG_CPU_BIG_ENDIAN
 	u16 hmo_ip_hdr_len;
 	u8 ip_nh_offset;
 	u8 options;
+#else
+	u8 options;
+	u8 ip_nh_offset;
+	u16 hmo_ip_hdr_len;
+#endif
 	union {
 		struct ipsec_decap_cbc cbc;
 		struct ipsec_decap_ctr ctr;
-- 
1.7.5.4

